### DISCUSSION TITLE
New ship: 🟢 Project status updates and issues side panel

### DISCUSSION THREAD

**Discussion by queenofcorgis** [community/community#86507] [Projects and Issues]
Today's changelog brings you project status updates and an updated issues side panel in Projects!

## :green_circle: Project status updates

You can now provide high level details on the status, timing, and progress of your project, directly from the project! This makes it easy to know and share with others how your work is progressing, any risks, and a history of when and why something changed, all in the same place where you're tracking your work.

You can access status updates from the `Project details` panel, where you can also add a short description or README with additional project information. Select <kbd>Add update</kbd> to give your project a `Status`, `Start date`, or `Target date`, along with additional details or mentioning another user or team. You can also edit, delete, or copy a link to a specific update to make it easy to share with others.

https://user-images.githubusercontent.com/101840513/297564154-0807fa59-b9fa-4ebb-b04a-7ce2e62e026d.mp4

Once you add a status update to a project, you'll find it visible in the project header and the project index pages, so you can quickly find and access the high level details for _all_ of your projects in a single place and drill in for more information. 

![project index page showing the description and start / end dates next to the project name](https://github.com/github/release-assets/assets/101840513/45dc2eff-feaa-48ef-ae44-bf79f0ca2866)

For more details, check out the [documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/sharing-project-updates).

## 🛝 Issues side panel in Projects

Projects has an updated issues side panel which matches the design of issues reached from a repository. In addition to providing a consistent experience, this update also means that issues accessed from a project have all timeline events and include any other projects the issue is a part of, making it easier to manage issues from either a project or a repository.   

![The image shows a project with an open issue page in a side panel](https://github.com/github/docs/assets/7584089/6d36c641-5643-422e-b600-ba29e84c4b71)

## :sparkles: Bug fixes and improvements
- Fixed a bug where labels were being unexpectedly changed on issues when adding them to a project
- Improved the workflow name editing experience by providing a dialog
- Improved keyboard focus and navigation on the Insights and Workflows pages

**Comment by mishmanners** [comment]
This is awesome!!! 🙌

**Comment by archon810** [comment]
Some bugs listed here: https://github.com/community/community/discussions/62400#discussioncomment-8175060.

**Comment by Faresabdelghany1** [comment]
nice

**Comment by linnforsman** [comment]
Perfect, this is what I wanted in GitHub projects.

**Comment by phifuh** [comment]
I like the design & this is a nice feature, but honestly this view should be optional. 
The clean minimalistic view, is on some issues/projects just better, at least for the issue comment history

**Comment by nvjustdev** [comment]
I am not able to get the status, the start and end dates to show up on the projects page as shown in the [image](https://private-user-images.githubusercontent.com/101840513/297564155-45dc2eff-feaa-48ef-ae44-bf79f0ca2866.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwNDgzNjAsIm5iZiI6MTcwNjA0ODA2MCwicGF0aCI6Ii8xMDE4NDA1MTMvMjk3NTY0MTU1LTQ1ZGMyZWZmLWZlYWEtNDhlZi1hZTQ0LWJmNzlmMGNhMjg2Ni5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyM1QyMjE0MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNWViMTQ4ZjE5MGRhOWYwZDM1MmFjNjJkZmNkOGE3MjJmYjhkOTEwOWZhNWY5MGRmODcyNDgwYTAyYjlkOGY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5AUdDMlSQWYDHIomyNT67iMXMotNXngw0Gbg89iavEc) 

Also is there a way to use APIs to get the project details like status, start and end date?

**Reply by rileybroughten** [reply]
@nvjustdev can you share more about this? You've added a status update with the `Status` and/or `Start date` and `Target date` fields to a project, and which projects pages can you not see the updates from (organization page, global github.com/projects, etc)?

**Reply by rileybroughten** [reply]
There isn't a way to get or set the status and date fields using the [GraphQL API](https://docs.github.com/en/graphql/reference/mutations) yet, but it's on our backlog as something we want to support 👍

**Reply by nvjustdev** [reply]
@rileybroughten i was looking at the organization's project (v2) list page which I understand as the project dashboard

**Reply by rileybroughten** [reply]
Have you added a status update to one or multiple projects? Let me know if you can provide any screenshots or a video walkthrough to help understand your setup and triage why you're not seeing the fields from the listing page.

**Reply by nvjustdev** [reply]
I did add for two projects. I will provide video or screenshots after i take off all proprietary info.

**Reply by nvjustdev** [reply]
<img width="1478" alt="Screenshot 2024-01-26 at 9 40 41 AM" src="https://github.com/community/community/assets/13179750/e4114c6d-8b32-4708-9dc6-ef926d31b9b7">

**Reply by rileybroughten** [reply]
See [comment](https://github.com/orgs/community/discussions/86507#discussioncomment-9906789) below 😁

**Comment by t3hero** [comment]
Would love to see this trigger the repo's webhooks.

**Comment by c00** [comment]
The Issue title in the sidebar is no longer a link to the issue (in a new tab) like it used to be (I used that a lot). Tho the number is clickable. Until you scroll down, then it becomes reversed. The title is clickable, but the number isn't.

Scrolled at the top:
![image](https://github.com/community/community/assets/6259056/d62cb415-3aee-42bc-a31b-9a5127fc0f04)


Scrolled down:
![image](https://github.com/community/community/assets/6259056/25d19d16-8efe-46c0-8a6d-64c9b0dfe8a6)

My vote goes towards both being clickable always.

**Comment by bewuethr** [comment]
This has uniformly elicited "this is so neat!" responses from everybody I showed it to. I'd love to be able to post these updates elsewhere, for example Slack, but I can't see an object for it in the GraphQL API. I can see a README object, but the status updates aren't in there.

Is that something that's planned (or maybe already available, but I overlooked it)?

**Reply by rileybroughten** [reply]
See [comment](https://github.com/orgs/community/discussions/86507#discussioncomment-9906789) below 😁

**Comment by SydAirey** [comment]
I think this is great!  Here is some feedback since recently using this feature...

**Project Boards used for actual Project Management:** 
- We currently use Project Boards for larger project initiatives and tracking.  I love that you can now add a project status and any important next steps in the side panel such as start date and due date.  My only request is to have the ability to take all of the current projects and turn that into more of a roadmap view based on the start date and targeted end date that is filled in. This could allow leadership to better understand all of the projects happening at the same time 

**Project Boards used for Sprint Planning**
- We currently use Project Boards for Sprint Planning and managing engineering work.  Is there any value in using this feature for scrum and kanban boards?  It could be useful to add any important status updates for the sprint and priority changes but lmk if anyone else has ideas.  Thanks!

**Comment by bxthings** [comment]
What happens to old status updates? Will they keep scrolling on this side panel forever, or drop off eventually, or is there a standalone page somewhere that they can be seen?

**Reply by rileybroughten** [reply]
Currently these are accessible from the side-panel where you provide and view project status updates, rather than a separate page. There is not yet a limit on how many can be viewed from this panel.

**Reply by phifuh** [reply]
Any thought on https://github.com/orgs/community/discussions/86507#discussioncomment-8185259?

**Comment by PearlHolland** [comment]
Discover the fastest and safest YouTube to MP3 Converter for your needs. Try [YTMP3](https://ytmp3.ing/) now!

**Comment by rileybroughten** [comment]
We've recently added GraphQL and webhooks support for project status updates 🚀 Check out the announcement and find more details in this week's [changelog](https://github.blog/changelog/2024-06-27-github-issues-projects-graphql-and-webhook-support-for-project-status-updates-and-more/)!

**Comment by rusaliya** [comment]
## 🐞 Bug report

@queenofcorgis,

Setting the Start/Target date seems to subtract 1 day from the _actual value_ once the update is saved.

![2024-07-23](https://github.com/user-attachments/assets/0aa67a53-e428-4437-aacb-745fa5710c96)

Is this a bug or am I missing something? Any assistance on how to resolve this bug is much appreciated. 🙂
Thank you in advance! 🙇‍♂️
